packages/compiler/src/extractor/extract.ts

```diff
@@ -49,7 +36,21 @@ export const extractProps = (
     ...defaultProps,
     ...propsMap,
   })) {
-    if (isStyledProp(propName.trim())) {
+    if (
+      typeof propValue === "string" &&
+      /[a-zA-Z]+\.[a-zA-Z0-9]+/.test(propValue)
+    ) {
+      const userTheme = theme.getUserTheme();
+      const propKey = propValue.split(".")[0] as Tokens;
+      if (userTheme[propKey] !== undefined) {
+        for (const key in userTheme[propKey]) {
+          if (propValue.trim() === key) {
+            styledProps[propName.trim()] = userTheme[propKey]![key];
+            break;
+          }
+        }
+      }
+    } else if (isStyledProp(propName.trim())) {
```

Comment:
@nojiritakeshi 
With this change, the HMR now works 🎉
However, I don't think it will work with Dynamic Props because the process has been moved from the system to the compiler.

```tsx
<Box p={12} color={() => "colors.blue.light"}>dynamic</Box>
```

I think it would be better to move this process to webpack-plugin and vite-plugin.

Reply:
I think this process needs to be implemented on the runtime side as well.
See the following line:

https://github.com/kuma-ui/kuma-ui/blob/7ebe9ed1120b9cd5cf451c1de124fab0abd49c6a/packages/core/src/components/Box/react/utils.ts#L30